platforms/chibios/drivers/wear_leveling/wear_leveling_efl.c

```diff
@@ -84,12 +84,14 @@ bool backing_store_init(void) {
 
     // Work out how many sectors we want to use, working backwards from the end of the flash
     flash_sector_t last_sector = desc->sectors_count - WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT;
+
+    // skip sectors that are past the actual flash size
+    while (flashGetSectorOffset(flash, last_sector) >= flash_size) {
```

Comment:
This code would trigger undefined behavior on the first iteration if EFL uses the `desc->sectors` array (e.g., on STM32F4xx), because the valid range of sector numbers is [0…`desc->sectors_count-1`].  The original code (before it had been broken by the change which added `WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT`) did not have this problem, because it subtracted at least 1 from `desc->sectors_count`.

Also should we guard against more pathological cases, like no free sectors, or even a sector being partially out of range of the reported flash size?  Although with the current error handling the only option is apparently to halt with a message that nobody would see.